### PR TITLE
fix: spec file validation now supports embedded structs

### PR DESCRIPTION
**Pull request by fridrik01**

### CODE REVIEW

File: config/spec/creator_test.go [hunk 1]
```diff
@@ -127,3 +130,60 @@ func TestCreateChainSpec_File(t *testing.T) {
...
+			name: "missing max-effective-balance after block-delay-configuration",
+			specContent: `[block-delay-configuration]
+max-block-delay = 300_000_000_000
+target-block-time = 2_000_000_000
+const-block-delay = 500_000_000
+consensus-update-height = 1
+consensus-enable-height = 2
+`,
+			wantErr: "missing required configuration for key: max-effective-balance",
+		},
```
Comment (calbera) [inline config/spec/creator_test.go:175]: can we also add a test case where err is nil? i.e. the valid way to load `block-delay-configuration`, as reference

**Review by calbera — APPROVED** [review]
utACK

**Review by 7fantasy7 — APPROVED** [review]

File: config/spec/creator_test.go (hunk 1 above)
Comment (fridrik01) [reply config/spec/creator_test.go:175]: @calbera good idea, updated the PR
